auth/principal.go

```diff
@@ -68,6 +68,10 @@ type Principal interface {
 	validate() error
 	setChannels(ch.TimedSet)
 	getVbNo(hashFunction VBHashFunction) uint16
+
+	// Cas value for the associated principal document in the bucket
+	Cas() uint64
```

Comment:
Could this use the https://godoc.org/github.com/couchbase/gocb#Cas type?

Reply:
This still needs to support walrus at this point.